src/Compilers/CSharp/Portable/BoundTree/DeconstructionLocalPendingInference.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class DeconstructionLocalPendingInference
+    {
+        public BoundLocal SetInferredType(TypeSymbol type, bool success)
```

Comment:
In the case of `success` should we assert that `type` is non-null
In the case of `!success` do we have any guarantese about `type`?  Is it null, error, etc ....